### DISCUSSION TITLE
Per user variables and folder organisation?

### DISCUSSION THREAD

**Discussion by xela-nosretap** [hoppscotch/hoppscotch#2122] [General]
1. Is there any way to create a truly local variable. We use a bearer token to auth with which is unique per browser instance. If someone changes the token in the team it changes for everyone and the requests stop working except for the last person that edited the variable.
2. Is there any way to re-organise the folders in a collection? I see I can drag and drop items but I don't seem to be able to rearrange folders

**Comment by liyasthomas** [comment]
Hi @alexander-paterson, thank you for trying out Hoppscotch.

1. I guess shared environments in Teams will resolve this feature request. You can create a team, invite members and create shared environment variables. We'll be adding this feature request to our roadmap. Watch this space for any further announcements on it.

2. We don't support re-organizing the folders/collections as of now. But would definitely see if it's a necessity and add it to our roadmap.

We appreciate you taking the time to request these features. Follow the project on GitHub to stay updated on future releases. Thank you.

**Comment by gu-maia** [comment]
Hi @liyasthomas and @alexander-paterson,

Any updates on this? My team is also using bearer tokens, which makes it difficult to use shared environments on Hoppscotch.

What we are currently doing is: In each personal workspace, we define the local environment with baseURL and bearer_token.

The problem is that when switching to any other environment (staging/production), we have to put our auth details in the request itself, overwriting everyone else's.

Any suggestions on how to handle this situation? Thanks!

**Comment by Icaruk** [comment]
How can I use shared variables like `API_URL` but use some variables like `TOKEN` locally?
Example:

- `API_URL`: shared with all users
- `TOKEN`: all users will use his own variable

**Comment by callum-atwal** [comment]
Any movement on this one? This would be a desirable feature for the same reasons listed above

**Comment by alecguintu** [comment]
Is there an update on this? We also have the same situation and wondering how to implement this somewhat "local" variable and would hopefully overwrite per local environment.

**Reply by liyasthomas** [reply]
Collections and Folders now support variables under [Collection Properties](https://docs.hoppscotch.io/documentation/features/collections#collection-properties). View various [Variable Scopes](https://docs.hoppscotch.io/documentation/features/variables#variable-scopes) to learn more.

**Reply by liyasthomas** [reply]
Secret variables’ values are stored only in active session storage, and they are local-only. For more information, refer to [Types of Variables in an Environment](https://docs.hoppscotch.io/documentation/features/environments#types-of-variables-in-an-environment).
